golang/Solutions/assgn1.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+var file string = "history.txt"
+
+func check(e error) {
+	if e != nil {
+		os.Truncate(file, 0)
+		panic(e)
+	}
+}
+
+func executeCmd(cmd string) string {
+	out, err := exec.Command(cmd).Output()
+	check(err)
+	return strings.TrimSuffix(string(out), "\n")
+}
+
+func addToHistory(cmd string) {
+	// write to file
+	f, err := os.OpenFile(file, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	check(err)
+	_, err = f.Write([]byte(cmd + "\n"))
+	check(err)
+	f.Close()
+}
+
+func main() {
+
+	var header string
+	var cmd string
+
+	for {
+		header = ""
+		header += executeCmd("whoami") + "/" + executeCmd("hostname") + executeCmd("pwd") + "~"
+		fmt.Print(header + " ")
+
+		reader := bufio.NewReader(os.Stdin)
+		read, _ := reader.ReadString('\n')
+		cmd = strings.Replace(read, "\n", "", -1)
+
+		if string(cmd[0]) != " " {
+			addToHistory(cmd)
+		} else {
+			cmd = strings.TrimSpace(cmd)
+		}
```

Comment:
Reason for this check? This won't add command to history if it's starting with space

Reply:
yeah this was intentional, history command also doesn't show the commands if space is added before it. Had checked it before implementing.